frontend/src/app/game/graphs/line-graph/line-graph.component.ts

```diff
@@ -104,20 +131,30 @@ export class LineGraphComponent implements OnInit, AfterViewInit {
         offset: 5,
         textAlign: 'center',
         rotation: -45,
-        backgroundColor: 'white',
-        borderColor: 'blue',
+        color: context => this.dataLabelNodes[context.dataIndex]?.event ? '#e3dce4' : '#676767',
+        backgroundColor: context => this.dataLabelNodes[context.dataIndex]?.event ? colors.event.background : 'white',
+        borderColor: colors.event.border,
+        borderWidth: context => this.dataLabelNodes[context.dataIndex]?.event ? 1 : 0,
         borderRadius: 3,
         padding: {
           top: 2,
           bottom: 2,
           right: 4,
           left: 4,
         },
-        display: context => Boolean(this.mitigationNodes[context.dataIndex]) ? 'auto' : false,
-        formatter: (_, context) => this.mitigationNodes[context.dataIndex],
+        display: context => this.showDataLabel(context.dataIndex),
+        formatter: (_, context) => {
+          const index = context.dataIndex;
+          let datalabel = this.dataLabelNodes[index]?.mitigations;
+          if (this.dataLabelNodes[index]?.event) {
+            datalabel = this.dataLabelNodes[index]?.event?.chosenMitigation?.label;
+          }
+          return datalabel;
+        },
         font: context => {
           const width = context.chart.width;
-          const size = Math.round(width! / 64);
+          let size = Math.round(width! / 64);
```

Comment:
Where did `64` come from?

Reply:
Only in my head, tested on various resolutions. Looks good.